scripts/integration-test.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+# prevent souring of this script, only allow execution
+$(return >/dev/null 2>&1)
+test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1; }
+          
+# set log id and use shared log function for readable logs
+declare mydir
+mydir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+declare HOPR_LOG_ID="hopr-connect-test"
+source "${mydir}/../scripts/utils.sh"
+
+# exit on errors, undefined variables, ensure errors in pipes are not hidden
+set -Eeuo pipefail
+
+usage() {
+  msg
+  msg "Usage: $0"
+  msg
+}
+
+# setup paths
+declare tmp="/tmp"
+declare server_log="${tmp}/hopr-connect-server.log"
+declare server_pid
+
+declare client0_log="${tmp}/hopr-connect-client0.log"
+declare client0_pid
+
+declare client1_log="${tmp}/hopr-connect-client1.log"
+declare client1_pid
+
+function free_ports {
+    for port in 9090 9091 9092; do
+        if lsof -i ":${port}" -s TCP:LISTEN; then
+        lsof -i ":${port}" -s TCP:LISTEN -t | xargs -I {} -n 1 kill {}
+        fi
+    done
+}
+
+function cleanup {
+  local EXIT_CODE=$?
+
+  trap - SIGINT SIGTERM ERR EXIT
+
+  # Cleaning up everything
+  log "Stopping up processes"
+  for pid in ${server_pid} ${client0_pid} ${client1_pid}; do
+    kill -9 ${pid}
+    log "killed ${pid}"    
+  done
+
+  free_ports
```

Comment:
why duplicate the effort you've done above? one should be enough

Reply:
Not sure what you mean. I kill the processes occupying ports on startup and in the exit trap, that's not duplication